src/Microsoft.DotNet.Arcade.Sdk/src/InstallDotNetTool.cs

```
@@ -0,0 +1,193 @@
+            }
+
+            string storePath = Path.Combine(ToolPath, ".store", Name.ToLowerInvariant());
+            string versionInstallPath = Path.Combine(storePath, version);
+
+            if (IsOrIsBeingInstalled(versionInstallPath))
+            {
+                return true;
+            }
+
+            FileStream lockFile;
+            try
+            {
+                lockFile = new FileStream(LockFilePath, FileMode.OpenOrCreate, FileAccess.Write, FileShare.None);
+            }
+            catch (IOException)
+            {
+                // Other process acquired the lock before we did
+                return IsOrIsBeingInstalled(versionInstallPath);
+            }
+
+            try
+            {
+                using (lockFile)
+                {
+                    Log.LogMessage($"Installing {Name} v{Version}...");
+
+                    var args = new List<string>()
+                    {
+                        "tool",
+                        "install",
+                        "--version",
+                        Version,
+                        "--tool-path",
+                        ToolPath,
+                    };
+
+                    if (!string.IsNullOrEmpty(Source))
+                    {
+                        args.Add("--add-source");
+                        args.Add(Source);
+                    }
+
+                    args.Add(Name);
+
+                    Command command = Command.Create(string.IsNullOrEmpty(DotnetPath) ? "dotnet" : DotnetPath, args);
+                    CommandResult result = command.Execute();
+
+                    if (result.ExitCode != 0)
+                    {
+                        return Directory.Exists(versionInstallPath);
+                    }
+
+                    Log.LogMessage($"{Name} v{Version} installed");
+
+                    return true;
+                }
+            }
+            finally
+            {
+                File.Delete(LockFilePath);
```

I think this may fail in some rare situations when other process gets a lock on the file before we manage to delete it. It should be enough to ignore any exception thrown by `Delete` in that case.